Senegal2019 revise

yining-sun opened the pull request:
- updated codes in 1.do- ANC
- updated codes in 2.do- delivery care
- updated 11.do- sleeping under bednet. Count both treated and nontreated as ITN. Recode mother education = 6 as missing.
- Changed codes in 7.do child vaccination to ensure kids 15-23 months old receiving measles2 are coded as measles1.
- Revised comments on Child Illness.

wengxyu1030 committed:
Refine the template

Made some small change to fix the code, there are cases where variables repeated generated several times. I add a capture condition to avoid the situation.

yining-sun committed:
Senegal2018 revise

- updated codes in 1.do- ANC
- updated codes in 2.do- delivery care - updated 11.do- sleeping under bednet. Count both treated and nontreated as ITN. Recode mother education = 6 as missing.
- Changed codes in 7.do child vaccination to ensure kids 15-23 months old receiving measles2 are coded as measles1.
- Revised comments on Child Illness.

yining-sun committed:
Senegal2019 revise

- updated codes in 1.do- ANC
- updated codes in 2.do- delivery care
- updated 11.do- sleeping under bednet. Count both treated and nontreated as ITN. Recode mother education = 6 as missing.
- Changed codes in 7.do child vaccination to ensure kids 15-23 months old receiving measles2 are coded as measles1.
- Revised comments on Child Illness.

yining-sun commented:
Dear both, 

I rebased this branch from the master before making the changes. However, it might still be unable to merge automatically because I didn't rebase when I first checked this survey weeks ago.

wengxyu1030 committed:
Merge pull request #13 from wengxyu1030/AW--template-enhancement

Refine the template

wengxyu1030 commented:
I see the changes you are making here are duplicated in Senegal 2018 #14 ,  @yining-sun please confirm. Please firstly follow my suggestions on #14 and pull request again for Senegal2018. Once this Senegal2018 is updated, the same code should work for Senegal 2019.

wengxyu1030 reviewed (changes requested):
Please first modify the Senegal2018.

wengxyu1030 commented on 13_adult.do:
```diff
@@ -46,19 +46,6 @@
 	replace a_bp_treat=1 if sh224==1 | sh324==1
 	}
 
-*a_bp_sys & a_bp_dial: 18y+ systolic & diastolic blood pressure (mmHg) in adult population 
```
Could you clarify the intention of the modification here? @yining-sun

yining-sun replied:
I didn't intend to delete this line. I will fix it.

yining-sun committed:
Senegal2018 revise

- collapsed the conditions for the same country.

yining-sun commented:
> I see the changes you are making here are duplicated in Senegal 2018 #14 , @yining-sun please confirm. Please firstly follow my suggestions on #14 and pull request again for Senegal2018. Once this Senegal2018 is updated, the same code should work for Senegal 2019.

Dear Aline, I have updated the Senegal2018. Shall I pull the request for Senegal2019 again after the pull request for Senegal2018 is updated?

wengxyu1030 committed:
Merge pull request #14 from wengxyu1030/YS-Senegal2018

Senegal2018 revise

wengxyu1030 commented:
> > I see the changes you are making here are duplicated in Senegal 2018 #14 , @yining-sun please confirm. Please firstly follow my suggestions on #14 and pull request again for Senegal2018. Once this Senegal2018 is updated, the same code should work for Senegal 2019.
> 
> Dear Aline, I have updated the Senegal2018. Shall I pull the request for Senegal2019 again after the pull request for Senegal2018 is updated?

Hi Yining, thanks for your feedback. I have reviewed your pull request of Senegal2018 and merged it to master. Please review if you have new updates in Senegal2019 comparing to Senegal2018 ( I have checked and to me it looks the updated template should work perfectly for Senegal2019), if you could confirm there's no further issue to be updated in Senegal2019, I will close this pull-request.

yining-sun committed:
Senegal2019 re-push

yining-sun committed:
Merge branch 'YS-Senegal2019' of https://github.com/wengxyu1030/DHS-Recode-VII into YS-Senegal2019

wengxyu1030 commented on 2_delivery_care.do:
```diff
@@ -95,30 +101,30 @@ gen country = regexs(1) if regexm(country_year, "([a-zA-Z]+)")
   	if !inlist(name,"Angola2015","Burundi2016","Cameroon2018","Colombia2015","Nepal2016","Tanzania2015"){
 	gen stay = 0 if m15 != .
 	replace stay = 1 if stay == 0 & (inrange(m61,124,198)|inrange(m61,201,298)|inrange(m61,301,399))
-	replace stay = . if inlist(m61,.,299,998)  & !inlist(m15,11,12,96) // filter question, based on m15
```
Hi @yining-sun  could you explain why did you delete the condition on missing?

wengxyu1030 replied:
Hi @yining-sun , thanks for the re-push. I am following up on this question, could you elaborate on your changes made here?

wengxyu1030 reviewed (changes requested):
Hi Yining, thanks for the update. I left some comments on specific code could you reply accordingly on the reasoning of the modifications?

wengxyu1030 commented on 2_delivery_care.do:
```diff
@@ -23,6 +23,12 @@ gen country = regexs(1) if regexm(country_year, "([a-zA-Z]+)")
 	replace `var' = . if !inlist(`var',0,1)
 	
 	 }
+	 if inlist(name, "Senegal2019") {
```
I am wondering why this is changed back to the older version instead of the latest one (the latest version in master), which I have agreed to your changes on Senegal2018 and merged it to master. I suspect it's the fetch and rebase practice is not done when you were modifying the template for Senegal2019?

yining-sun replied:
Hi Aline, I actually did the fetch, and then rebase from the master before opening these files. There were several conflicts after rebasing, so I adopted the changes from the master branch manually. I don't know why it changes back to the older version. 

I have just rebased and then changed the codes again for re-push.

wengxyu1030 replied:
Hi Yining, thanks for the clarification, it's great that you fetched and re-based, they are indeed good practices. It is understandable the conflicts came up because the master branch is updated between your two fetches/re-bases. 

I am thinking this issue might be caused by the accident that when you re-base and solving the conflicts manually, you probably accidentally discarded the updated master version instead kept your older version. I am attaching this guide note https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/resolving-a-merge-conflict-on-github , which might give you more sense about how to resolve conflict manually.

wengxyu1030 committed:
Update the code to be consistent to the master 

Aline try to manually update the branch to be consistent with the master branch.

yining-sun committed:
Senegal2019 repush

resolve the rebasing conflicts

yining-sun committed:
Update 7_child_vaccination.do

wengxyu1030 commented:
@yining-sun created another branch where the rebase issue is solved in that branch #16 , thereby I am closing this issue and pull request.

robin-wang commented:
@wengxyu1030 Thank you!
